bouncer/tests/delta_base_ingress.ts

```diff
@@ -0,0 +1,184 @@
+import { killEngines, startEngines } from '../shared/upgrade_network';
+
+let swapsWitnessed: number = 0;
+
+async function deltaBasedIngressTest(
+  // Directory where the node and CFE binaries of the new version are located
+  binariesPath: string,
+  localnetInitPath: string,
+  numberOfNodes: 1 | 3 = 1,
+): Promise<void> {
+  const sourceAsset = 'Sol';
+  const amountFirstDeposit = '5';
+  const amountSecondDeposit = '1';
+
+  // Monitor swap events to make sure there is only one
+  let swapScheduledHandle = observeEvent('swapping:SwapScheduled', {
+    test: (event) => {
+      const data = event.data;
+
+      swapsWitnessed++;
+      console.log('Swap Scheduled found, swaps witnessed: ', swapsWitnessed);
+
+      if (swapsWitnessed > 1) {
+        throw new Error('More than one swaps were initiated');
+      }
+
+      const inputAmount = Number(data.inputAmount.replace(/,/g, ''));
+      if (
+        inputAmount > Number(amountToFineAmount(amountFirstDeposit, assetDecimals(sourceAsset)))
+      ) {
+        throw new Error(
+          'Swap input amount is greater than the first deposit ' + inputAmount.toString(),
+        );
+      }
+      return false;
+    },
+    abortable: true,
+  });
+
+  // Initiate swap from Solana
+  const swapParams = await testSwap(
+    sourceAsset,
+    'Eth',
+    undefined,
+    undefined,
+    undefined,
+    ' DeltaBasedIngress',
+    amountFirstDeposit.toString(),
+  );
+
+  await observeFetch(sourceAsset, swapParams.depositAddress);
+
+  await killEngines();
+  await startEngines(localnetInitPath, binariesPath, numberOfNodes);
+
+  // Wait to ensure no new swap is being triggered after restart.
+  console.log('Waiting for 30 seconds to ensure no swap is being triggered after restart');
+  await sleep(30000);
+  swapScheduledHandle.stop();
+
+  if (swapsWitnessed !== 1) {
+    throw new Error('No swap was initiated. Swaps witnessed: ' + swapsWitnessed);
+  }
+
+  // Kill the engine
+  await killEngines();
+
+  // Start another swap doing another deposit to the same address
+  const swapHandle = doPerformSwap(
+    swapParams,
+    `[${sourceAsset}->Eth DeltaBasedIngressSecondDeposit]`,
+    undefined,
+    undefined,
+    amountSecondDeposit,
+  );
+
+  swapScheduledHandle = observeEvent('swapping:SwapScheduled', {
+    test: (event) => {
+      const data = event.data;
+
+      swapsWitnessed++;
+      console.log('Swap Scheduled found, swaps witnessed: ', swapsWitnessed);
+
+      if (swapsWitnessed > 2) {
+        throw new Error('More than two swaps were initiated');
+      }
+      const inputAmount = Number(data.inputAmount.replace(/,/g, ''));
+      if (
+        inputAmount > Number(amountToFineAmount(amountSecondDeposit, assetDecimals(sourceAsset)))
+      ) {
+        throw new Error(
+          'Swap input amount is greater than the second deposit ' + inputAmount.toString(),
+        );
+      }
+      return false;
+    },
+    abortable: true,
+  });
```

Comment:
This is almost identical to the above call, we can make a local function to dedupe